### PR TITLE
OEL-2794: Update components to run on Drupal 10.2.x 

### PR DESCRIPTION
**Pull request by AaronGilMartinez**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
         "drupal/block_field": "^1.0@RC",
         "drupal/ckeditor": "^1.0",
         "drupal/composite_reference": "^2.2",
-        "drupal/core-recommended": "^10.1",
+        "drupal/core-recommended": "^10.2",
```
Comment (drishu) [inline composer.json]: try with ^10 and run composer update for these packages with the dependencies flag, it should install the latest 10.2

also I think the php requirement should be a minimum of php 8.1

you can remove oe_starter_content and oe_content_extra from the "repositories" section while we are here, they where added to packagist

File: tests/src/ExistingSiteJavascript/ShowcaseExistingSiteJavascriptTestBase.php [hunk 2]
```diff
@@ -48,6 +48,7 @@ public function tearDown(): void {
     $this->restoreConfiguration();
 
     parent::tearDown();
+
```
Comment (drishu) [inline tests/src/ExistingSiteJavascript/ShowcaseExistingSiteJavascriptTestBase.php]: remove

File: .drone.yml [hunk 3]
```diff
@@ -114,7 +112,7 @@ pipeline:
     image: registry.fpfis.eu/fpfis/httpd-php:8.1-ci
     commands:
       - ./vendor/bin/run ci:site-setup
-      - ./vendor/bin/drush en -y media_avportal_mock oe_media_oembed_mock
+      - ./vendor/bin/drush en -y media_avportal_mock oe_media_oembed_mock js_testing_ajax_request_test
```
Comment (drishu) [inline .drone.yml:115]: intresting but I dont see the usage, can you explain to me a bit?

File: modules/oe_showcase_list_pages/tests/src/ExistingSiteJavascript/ListPagesTest.php [hunk 4]
```diff
@@ -948,4 +937,15 @@ protected function scrollIntoView(string $selector): void {
     $this->getSession()->executeScript("window.scrollBy({top: document.querySelector('$selector').offsetTop, left: 0, behavior: 'instant'})");
   }
 
+  /**
+   * Select a sort option and waits until is applied.
+   *
+   * @param string $option
+   *   The option to be applied.
+   */
+  protected function sortByOption(string $option): void {
```
Comment (drishu) [inline modules/oe_showcase_list_pages/tests/src/ExistingSiteJavascript/ListPagesTest.php]: the name of the function does not reflect what it does, this function selects an option, so selectSortByOption

**Review by drishu — CHANGES_REQUESTED** [review]
We can also safely drop version 9 from all core_version_requirement

File: modules/oe_showcase_list_pages/tests/src/ExistingSiteJavascript/ListPagesTest.php (hunk 4 above)
Comment (AaronGilMartinez) [reply modules/oe_showcase_list_pages/tests/src/ExistingSiteJavascript/ListPagesTest.php]: Renamed

File: tests/src/ExistingSiteJavascript/ShowcaseExistingSiteJavascriptTestBase.php (hunk 2 above)
Comment (AaronGilMartinez) [reply tests/src/ExistingSiteJavascript/ShowcaseExistingSiteJavascriptTestBase.php]: done

File: composer.json (hunk 1 above)
Comment (AaronGilMartinez) [reply composer.json]: done

File: .drone.yml (hunk 3 above)
Comment (AaronGilMartinez) [reply .drone.yml:115]: The core Javascript tests base class (core/tests/Drupal/FunctionalJavascriptTests/WebDriverTestBase.php) uses that module to track Ajax requests, with Drupal 10.2 there are changes and new checks on the function assertWaitOnAjaxRequest (drupal-site/web/core/tests/Drupal/FunctionalJavascriptTests/JSWebAssert.php) and if there is no Ajax request the test fails.
Our Javascript tests extends the provided by weitzman/drupal-test-site so the module is not installed and some test fail.

**Review by drishu — DISMISSED** [review]

**Review by drishu — APPROVED** [review]

File: composer.json [hunk 5]
```diff
@@ -112,7 +105,7 @@
                 "Allow altering the post_update run order @see https://www.drupal.org/project/drupal/issues/3129231": "https://www.drupal.org/files/issues/2022-06-27/drupal-post_update_alter-3129231-20.patch",
                 "Unserialize(): Passing null to parameter #1 ($data) of type string is deprecated @see https://www.drupal.org/project/drupal/issues/3300404": "https://www.drupal.org/files/issues/2023-03-03/3300404-28-map-field-failing-php-81-combined.patch",
                 "strpos(): Passing null is deprecated in core/lib/Drupal/Core/Mail/Plugin/Mail/PhpMail.php @see https://www.drupal.org/project/drupal/issues/3328187": "https://www.drupal.org/files/issues/2023-07-05/3372667-2_0.patch",
-                "Simple decimals fail to pass validation @see https://www.drupal.org/project/drupal/issues/2230909": "https://www.drupal.org/files/issues/2022-11-19/2230909-285-D10.patch"
+                "Simple decimals fail to pass validation @see https://www.drupal.org/project/drupal/issues/2230909": "https://www.drupal.org/files/issues/2023-12-21/2230909-309.patch"
```
Comment (drishu) [inline composer.json]: think we should use the stable oe_content version and drop this line

**Review by drishu — DISMISSED** [review]

File: composer.json (hunk 5 above)
Comment (AaronGilMartinez) [reply composer.json]: Removed and updated

**Review by brummbar — APPROVED** [review]

**Pull request merged**